ops.qmd

```diff
@@ -2,84 +2,760 @@
+
+Data scientists drive model creation, improvement and innovation through their expertise in ML techniques. They collaborate closely with other roles to ensure models create maximum business impact.
+
+### ML Engineers
+
+ML engineers enable models data scientists develop to be productized and deployed at scale. Their expertise makes models reliably serve predictions in applications and business processes. Their main responsibilities include:
+
+* Taking prototype models from data scientists and hardening them for production environments through coding best practices.
+* Building APIs and microservices for model deployment using tools like Flask, FastAPI. Containerizing models with Docker.
+* Managing model versions and sinaging new models into production using CI/CD pipelines. Implementing canary releases, A/B tests, and rollback procedures.
+* Optimizing model performance for high scalability, low latency and cost-efficiency. Leveraging compression, quantization, multi-model serving.
+* Monitoring models once in production and ensuring continued reliability and accuracy. Retraining models periodically.
+
+For example, a ML engineer may take a TensorFlow fraud detection model developed by data scientists and containerize it using TensorFlow Serving for scalable deployment. The model would be integrated into the company's transaction processing pipeline via APIs. The ML engineer implements a model registry and CI/CD pipeline using MLFlow and Jenkins to reliably deploy model updates. The ML engineers would then monitor the running model for continued performance using tools like Prometheus and Grafana. If model accuracy drops, they initiate retraining and deployment of a new model version.
+
+The ML engineering team enables data science models to progress smoothly into sustainable and robust production systems. Their expertise in building modular, monitored systems delivers continuous business value.
+
+### DevOps Engineers
+
+DevOps engineers enable MLOps by building and managing the underlying infrastructure for developing, deploying, and monitoring ML models. They provide the cloud architecture and automation pipelines. Their main responsibilities include:
+
+* Provisioning and managing cloud infrastructure for ML workflows using IaC tools like Terraform, Docker, Kubernetes. 
+* Developing CI/CD pipelines for model retraining, validation, and deployment. Integrating ML tools into the pipeline like MLflow, Kubeflow.
+* Monitoring model and infrastructure performance using tools like Prometheus, Grafana, ELK stack. Building alerts and dashboards.
```

Comment:
Possible link for ELK stack: https://aws.amazon.com/what-is/elk-stack/#:~:text=offered%20by%20AWS%3F-,What%20is%20the%20ELK%20Stack%3F,Elasticsearch%2C%20Logstash%2C%20and%20Kibana.

Reply:
Done!